brigade.js

```diff
@@ -99,6 +103,43 @@ events.on("error", (e) => {
     console.log("==> Event " + e.type + " caused by " + e.provider + " cause class" + e.cause + e.cause.reason)
 })
 
-events.on("after", (e) => {  
-    console.log("After event fired " + util.inspect(e, false, null) )
+events.on("after", (e, p) => {
```

Comment:
the only concern with the "after" event is it happens no matter the success/failure of the other Jobs. quick'n'dirty fix could be to set a global var success = false and set to true within Promise results of the robot job

Reply:
Agreed. Added in logic to see if the test_results are populated. If they aren't then the slack job won't run. This should handle the case when the robot job fails because it grabs the results after a successful run of the job.